### ISSUE TITLE
Missing symbol glyphs from default font set

### ISSUE THREAD

**Issue by sirkitree** [Frooxius/NeosPublic#352]
I have used a few various font symbols in my UI, but they are breaking (just showing [] ) since the font update. I've tried pulling in a few different fonts that support these glyphs but they're still showing as [].

![20191226101340_1](https://user-images.githubusercontent.com/28543/71485987-d513ab80-27c8-11ea-8057-f923db38451a.jpg)
You can see the glyph in the Inspector renders just fine. This was taken from the following website: https://www.w3schools.com/charsets/ref_utf_geometric.asp

However, it renders like so:
![20191226101352_1](https://user-images.githubusercontent.com/28543/71486014-00969600-27c9-11ea-83e5-409c5ce55c82.jpg)

Wehn inspecting the website in which this is rendering, it's using the font 'Verdana', so I know that Verdana supports this glyph. If I import that into Neos and utilize it in the 'Font' property, I still get []. 

Am I misunderstanding how this is supposed to be used? Or is there further work needed to support such text icons?

**Comment by sirkitree** [comment]
I also tried pulling in the webdings and wingdings fonts to see if those worked, but got squares again.

[dingFonts.zip](https://github.com/Frooxius/NeosPublic/files/4005857/dingFonts.zip)

**Comment by Frooxius** [comment]
UTF-8 is already supported on the wire, but that's not how the strings are represented in the memory. CLR uses UTF-16 to represent the Unicode codepoints. It's possible that those symbols use surrogate pairs? Can you send me some of them here so I can have a look?

The text shaping process currently doesn't handle surrogate pairs, so it's possible it doesn't pick the right glyphs for those.

**Comment by sirkitree** [comment]
These are pretty standard fonts that should be available in your C:/Windows/Fonts
I've tried the dings I attached above, Verdana, and Microsoft sans serif regular

**Comment by Frooxius** [comment]
I don't need the fonts, I need the characters you're using. I need to check if they're using surrogate pairs for representation. I'm checking with some other ones that I found, but it would help knowing the symbols you had trouble with.

**Comment by sirkitree** [comment]
Ah, my appologies, I misunderstood. Here are a few:
![20191227094402_1](https://user-images.githubusercontent.com/28543/71526871-ad3a4b80-288d-11ea-9504-10dc16346a65.jpg)


* ▶ | 9654 | 25B6 |   | BLACK RIGHT-POINTING TRIANGLE
* ↤ | 8612 | 21A4 |   | LEFTWARDS ARROW FROM BAR
* ⊗ | 8855 | 2297 | &otimes; | CIRCLED TIMES

**Comment by Frooxius** [comment]
Thanks! I checked those and they do not use surrogate pairs. However the current fonts and the ones you tried simply don't have those glyphs, so they don't render. 

The Noto family includes two symbol fonts which include some of these though, I'll include them by default in Neos so the symbols should start working. 

The only one that I can't find in any of the fonts yet is ↤ | 8612 | 21A4 |   | LEFTWARDS ARROW FROM BAR

**Comment by sirkitree** [comment]
In inspecting this webpage, it looks like it's using the following:
`-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;`

Also, if the answer is that I should just find different icons, or I should do something differently, that is ok too. Perhaps glyphs are not what I should sue but actual svg icons (though I'd have to convert those to geometry since #137 is not a thing yet)

In general, if it shows up in a browser I was expecting them to just work (as this is the list I'm going from) but it sounds like it's much more complex than that. Happy to go with whatever you recommend without making this something you absolutely need to support.

**Comment by Frooxius** [comment]
I found newer versions of the Noto fonts, including a new one Noto Math which seems to contain all those symbols. They will be included in the next build of Neos, so you don't have to do anything.

It works in the browser, because it uses lots of font files for the missing glyphs in the background. In Neos all the fonts have to be explicitly provided.

**Comment by Frooxius** [comment]
Fixed in 0.8.19.36896, the build now includes Noto Symbols 2 and Noto Math fonts as fallbacks, which include all the symbols you've had trouble with (and many more of course).

I've also fixed handling of surrogate pairs, although that wasn't causing issue in this particular case.

However webdings and wingdings still don't work. Problem with those is that they're not Unicode fonts, but have a special character map instead. Given how old they are and that Unicode symbols pretty much supersede them, I'm not adding support for that right now. If they're really important to your usecase, feel free to open another issue for them. 

Thanks for the report and all your help!
